src/app/dashboard/dashboard.component.ts

```diff
@@ -173,7 +173,14 @@ export class DashboardComponent implements OnInit {
       this.search.competencies.map(async (comp: Competency) => {
         await this.competencyService.getCompetencyById(comp._id)
           .then(async (comp: Competency) => {
-            this.loadedCompetencies.push(comp);
+            // apply work role filter if selected else load all
+            if (this.selected.work_role.length > 0) {
+              if (this.selected.work_role.includes(comp.behavior.work_role)) {
+                this.loadedCompetencies.push(comp);
+              }
+            } else {
+              this.loadedCompetencies.push(comp);
+            }
```

Comment:
```suggestion
            this.loadedCompetencies.push(comp);
```
The API should handle this filtering for us. We would pass the search query and that would return the results list with the correct competencies. Also, you should be able to select more than one workrole, it would be an aggregation of competencies with either workrole

Reply:
The API can now handle the filtering.
[111](https://github.com/Cyber4All/competency-api/pull/111).
Selecting multiple workrole is possible also.